In [1]:
import pandas as pd
import matplotlib as plt

In [13]:
#carga del arhivo movies.

df = pd.read_csv("movies_dataset.csv")
df.head()

C:\Users\fmgc1\AppData\Local\Temp\ipykernel_5292\215091187.py:3: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("movies_dataset.csv")


adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

De un primer analisis rapido y consultando el diccionario de datos pudimos identificar algunas columnas que consideramos que no aportarian informacion util al algoritmo de recomendacion. Por ejemplo la columna homepage contiene las direcciones web de las paginas de las peliculas, la columna overview contiene una reseña del contenido de la pelicula.

In [15]:
#Eliminamos las columnas que consideramos que no aportarian informacion util
df = df.drop(columns=['homepage', 'imdb_id','overview', 'status','overview', 'poster_path','tagline','video'])
df.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres     id    imdb_id  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862  tt0114709   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844  tt0113497   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602  tt0113228   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357  tt0114885   
4                     [{'id': 35, 'name': 'Comedy'}]  11862  tt0113041   

  original_language               original_title popularity  \
0                en                    Toy Story  21.946943   
1                en                      Jumanji  17.015539   
2                en             Grumpier Old Men    11.7129   
3                en            Waiting to Exhale   3.859495   
4                en  Father of the Bride Part II   8.387519   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

                         title  vote_average  vote_count  
0                    Toy Story           7.7      5415.0  
1                      Jumanji           6.9      2413.0  
2             Grumpier Old Men           6.5        92.0  
3            Waiting to Exhale           6.1        34.0  
4  Father of the Bride Part II           5.7       173.0

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   id                     45466 non-null  object 
 5   imdb_id                45449 non-null  object 
 6   original_language      45455 non-null  object 
 7   original_title         45466 non-null  object 
 8   popularity             45461 non-null  object 
 9   production_companies   45463 non-null  object 
 10  production_countries   45463 non-null  object 
 11  release_date           45379 non-null  object 
 12  revenue                45460 non-null  float64
 13  runtime                45203 non-null  float64
 14  spoken_languages       45460 non-null  object 
 15  ti

Notamos que de de las 45466 filas totales la columna title contiene 6 valores nulos, entonces los analizamos

In [17]:
df[df['title'].isna()]

adult  \
19729                                              False   
19730                                 - Written by Ørnås   
29502                                              False   
29503   Rune Balot goes to a casino connected to the ...   
35586                                              False   
35587   Avalanche Sharks tells the story of a bikini ...   

                                   belongs_to_collection  \
19729                                                NaN   
19730                                           0.065736   
29502  {'id': 122661, 'name': 'Mardock Scramble Colle...   
29503                                           1.931659   
35586                                                NaN   
35587                                           2.185485   

                                 budget  \
19729                                 0   
19730  /ff9qCepilowshEtG2GYWwzt2bs4.jpg   
29502                                 0   
29503  /zV8bHuSL6WXoD6FWogP9j4x80bL.jpg   
35586                                 0   
35587  /zaSf5OG7V8X8gqFvly88zDdRm46.jpg   

                                                  genres          id  \
19729  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...       82663   
19730  [{'name': 'Carousel Productions', 'id': 11176}...  1997-08-20   
29502  [{'id': 16, 'name': 'Animation'}, {'id': 878, ...      122662   
29503  [{'name': 'Aniplex', 'id': 2883}, {'name': 'Go...  2012-09-29   
35586  [{'id': 10770, 'name': 'TV Movie'}, {'id': 28,...      249260   
35587  [{'name': 'Odyssey Media', 'id': 17161}, {'nam...  2014-01-01   

         imdb_id original_language                            original_title  \
19729  tt0113002                en                              Midnight Man   
19730          0             104.0  [{'iso_639_1': 'en', 'name': 'English'}]   
29502  tt2423504                ja                          マルドゥック・スクランブル 排気   
29503          0              68.0      [{'iso_639_1': 'ja', 'name': '日本語'}]   
35586  tt2622826                en                          Avalanche Sharks   
35587          0              82.0  [{'iso_639_1': 'en', 'name': 'English'}]   

                  popularity production_companies production_countries  \
19729                    NaN                  NaN                  NaN   
19730                    NaN                False                  6.0   
29502                    NaN                  NaN                  NaN   
29503                    NaN                False                  7.0   
35586                    NaN                  NaN                  NaN   
35587  Beware Of Frost Bites                False                  4.3   

      release_date  revenue  runtime spoken_languages title  vote_average  \
19729          NaN      NaN      NaN              NaN   NaN           NaN   
19730            1      NaN      NaN              NaN   NaN           NaN   
29502          NaN      NaN      NaN              NaN   NaN           NaN   
29503           12      NaN      NaN              NaN   NaN           NaN   
35586          NaN      NaN      NaN              NaN   NaN           NaN   
35587           22      NaN      NaN              NaN   NaN           NaN   

       vote_count  
19729         NaN  
19730         NaN  
29502         NaN  
29503         NaN  
35586         NaN  
35587         NaN

Notamos que podemos eliminar del dataset estas 6 filas

In [19]:
# elimino los valores nan de la columna title
df = df.dropna(subset=['title'])

Analizamos ahora la columna adult

In [20]:
df['adult'].value_counts()

adult
False    45451
True         9
Name: count, dtype: int64

Notamos 9 filas con valor TRUE las analizamos en detalle

In [21]:
df[df['adult'] == "True"]

adult belongs_to_collection  budget  \
19489  True                   NaN       0   
28701  True                   NaN       0   
31934  True                   NaN       0   
32113  True                   NaN       0   
39901  True                   NaN       0   
39902  True                   NaN       0   
40574  True                   NaN       0   
41009  True                   NaN  750000   
43090  True                   NaN       0   

                                                  genres      id    imdb_id  \
19489                     [{'id': 27, 'name': 'Horror'}]    5422  tt0079642   
28701                      [{'id': 80, 'name': 'Crime'}]  169158  tt0119931   
31934                     [{'id': 35, 'name': 'Comedy'}]   44781  tt0322232   
32113  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  324230  tt3094816   
39901  [{'id': 80, 'name': 'Crime'}, {'id': 27, 'name...   35731  tt1161951   
39902  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   55774  tt1153101   
40574  [{'id': 80, 'name': 'Crime'}, {'id': 27, 'name...   39922  tt0070696   
41009  [{'id': 53, 'name': 'Thriller'}, {'id': 80, 'n...  376004  tt4044464   
43090  [{'id': 878, 'name': 'Science Fiction'}, {'id'...  448290  tt2697688   

      original_language                       original_title popularity  \
19489                it  Le notti erotiche dei morti viventi   1.420104   
28701                en                             Standoff   1.349745   
31934                cn                                發電悄嬌娃   0.001803   
32113                es                          Diet of Sex   6.889832   
39901                en           Amateur Porn Star Killer 2   0.587751   
39902                en                             The Band   0.417214   
40574                da                              Dværgen   1.182635   
41009                en                           Adulterers    1.13377   
43090                en                           Half -Life    0.28268   

                                    production_companies  \
19489             [{'name': 'Stefano Film', 'id': 6117}]   
28701         [{'name': 'Trimark Pictures', 'id': 4063}]   
31934                                                 []   
32113              [{'name': 'KeyAtomics', 'id': 46055}]   
39901         [{'name': 'Alter Ego Cinema', 'id': 5639}]   
39902                                                 []   
40574           [{'name': 'Scandica Film', 'id': 48812}]   
41009                                                 []   
43090  [{'name': 'Bad Robot', 'id': 11461}, {'name': ...   

                                    production_countries release_date  \
19489            [{'iso_3166_1': 'IT', 'name': 'Italy'}]   1980-11-18   
28701  [{'iso_3166_1': 'US', 'name': 'United States o...   1998-01-11   
31934        [{'iso_3166_1': 'HK', 'name': 'Hong Kong'}]   2001-04-26   
32113            [{'iso_3166_1': 'ES', 'name': 'Spain'}]   2014-02-14   
39901  [{'iso_3166_1': 'US', 'name': 'United States o...   2008-05-13   
39902        [{'iso_3166_1': 'AU', 'name': 'Australia'}]   2009-11-17   
40574  [{'iso_3166_1': 'DK', 'name': 'Denmark'}, {'is...   1973-12-20   
41009                                                 []   2016-01-05   
43090  [{'iso_3166_1': 'US', 'name': 'United States o...          NaN   

       revenue  runtime                            spoken_languages  \
19489      0.0    112.0   [{'iso_639_1': 'it', 'name': 'Italiano'}]   
28701      0.0     91.0    [{'iso_639_1': 'en', 'name': 'English'}]   
31934      0.0     89.0  [{'iso_639_1': 'cn', 'name': '广州话 / 廣州話'}]   
32113      0.0     72.0    [{'iso_639_1': 'es', 'name': 'Español'}]   
39901      0.0      0.0    [{'iso_639_1': 'en', 'name': 'English'}]   
39902      0.0     90.0    [{'iso_639_1': 'en', 'name': 'English'}]   
40574      0.0     92.0    [{'iso_639_1': 'en', 'name': 'English'}]   
41009      0.0     80.0    [{'iso_639_1': 'en', 'name': 'English'}]   
43090      0.0      0.0    [{'iso_639_1': 

se decide hacer un recomendador "family friendly" por lo que nos quedamos con los titulos que no son para adultos

In [22]:
df = df[df['adult'] == "False"]

Una vez echo esto la columna adult queda con todos los valores "False" por lo que se puede quitar esta columna

In [23]:
#comprobamos
df['adult'].value_counts()

adult
False    45451
Name: count, dtype: int64

In [24]:
#eliminamos la columna
df = df.drop('adult', axis=1)

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45451 entries, 0 to 45465
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   belongs_to_collection  4490 non-null   object 
 1   budget                 45451 non-null  object 
 2   genres                 45451 non-null  object 
 3   id                     45451 non-null  object 
 4   imdb_id                45434 non-null  object 
 5   original_language      45440 non-null  object 
 6   original_title         45451 non-null  object 
 7   popularity             45451 non-null  object 
 8   production_companies   45451 non-null  object 
 9   production_countries   45451 non-null  object 
 10  release_date           45368 non-null  object 
 11  revenue                45451 non-null  float64
 12  runtime                45194 non-null  float64
 13  spoken_languages       45451 non-null  object 
 14  title                  45451 non-null  object 
 15  vote_av

In [27]:
df.head()

belongs_to_collection    budget  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                NaN  65000000   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                NaN  16000000   
4  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres     id    imdb_id  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862  tt0114709   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844  tt0113497   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602  tt0113228   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357  tt0114885   
4                     [{'id': 35, 'name': 'Comedy'}]  11862  tt0113041   

  original_language               original_title popularity  \
0                en                    Toy Story  21.946943   
1                en                      Jumanji  17.015539   
2                en             Grumpier Old Men    11.7129   
3                en            Waiting to Exhale   3.859495   
4                en  Father of the Bride Part II   8.387519   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

                         title  vote_average  vote_count  
0                    Toy Story           7.7      5415.0  
1                      Jumanji           6.9      2413.0  
2             Grumpier Old Men           6.5        92.0  
3            Waiting to Exhale           6.1        34.0  
4  Father of the Bride Part II           5.7       173.0